### PR TITLE
Add deployment usage section

### PR DESCRIPTION
**Pull request by ironcladlou**
This is ported from the [origin repo docs](https://github.com/openshift/origin/blob/master/docs/deployments.md). I changed things up a bit to take advantage of AsciiDoc features. Please review.

### CODE REVIEW

**Comment by ironcladlou** [comment]
cc @pmorie @nhr

File: using_openshift/deployments.adoc [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+:icons:
+:experimental:
+:toc: macro
+:toc-title: 
+
+toc::[]
+
+== Overview
+
+In OpenShift, deployment is an update to a single replication controller's pod template based on triggered events. The deployment subsystem provides:
```
Comment (nhr) [inline using_openshift/deployments.adoc]: "In OpenShift, _a_ deployment..."

**Comment by nhr** [comment]
Pedantic edit aside, this looks great.

File: using_openshift/deployments.adoc (hunk 1 above)
Comment (ironcladlou) [inline using_openshift/deployments.adoc]: Fixed, thanks!



----- Original Message -----

**Comment by nhr** [comment]
Confirmed fix, merging.

**Pull request merged**